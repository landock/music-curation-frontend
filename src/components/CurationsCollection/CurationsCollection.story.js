import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import CurationsCollection from '.';

const curations = [
  {
    id: '23',
    imageUrl: 'https://placeimg.com/250/250/tech/grayscale',
    tracks: [
      {
        id: '12',
        imageUrl: 'https://placeimg.com/250/250/animals',
        artistName: 'spongebop',
        trackName: 'squarewave',
      },
    ],
    name: 'story testing',
  },
  {
    id: '54',
    imageUrl: 'https://placeimg.com/250/250/animals/grayscale',
    tracks: [
      {
        id: '12',
        imageUrl: 'https://placeimg.com/250/250/tech',
        artistName: 'Doug Handleman',
        trackName: 'Bing bong eats fresh fish',
      },
    ],
    name: 'Space paper & Beezlebub',
  },
];
storiesOf('CurationsCollection', module).add('default', () => (
  <CurationsCollection curations={{ curations }} />
));
