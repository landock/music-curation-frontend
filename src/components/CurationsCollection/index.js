import React, { Component } from 'react';

class CurationsCollection extends Component {
  render() {
    const { curations } = this.props;
    const hasCurations =
      curations && curations.curations && curations.curations[0];
    const curationsMarkup = hasCurations ? (
      curations.curations.map(curation => (
        <div key={curation.id}>
          <h1>
            {`${curation.name} - ${curation.id}`}
            <img style={{ height: '120px' }} src={curation.imageUrl} />
          </h1>
          {curation.tracks &&
            curation.tracks.map(track => {
              return (
                <div key={track.id}>
                  <img style={{ height: '120px' }} src={track.imageUrl} />
                  <p>{`${track.artistName} - ${track.trackName}`}</p>
                </div>
              );
            })}
        </div>
      ))
    ) : (
      <p>No Curations Found</p>
    );

    debugger;
    return curationsMarkup;
  }
}

export default CurationsCollection;
