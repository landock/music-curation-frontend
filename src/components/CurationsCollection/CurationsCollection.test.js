import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import CurationsCollection from '.';

describe('CurationsCollection', () => {
  let component;
  it('Render without crashing', () => {
    component = renderer.create(<CurationsCollection />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render placeholder message if no curations are passed as props', () => {
    component = shallow(<CurationsCollection />);
    const placeholderMarkup = <p>No Curations Found</p>;
    expect(component.contains(placeholderMarkup)).toBe(true);
  });
});
