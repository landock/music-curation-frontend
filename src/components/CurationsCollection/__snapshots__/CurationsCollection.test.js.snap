// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurationsCollection Render without crashing 1`] = `
<p>
  No Curations Found
</p>
`;
